test/OpenTelemetry.Tests/Metrics/AggregatorTest.cs

```diff
@@ -188,6 +188,42 @@ public void HistogramWithOnlySumCount()
             Assert.False(enumerator.MoveNext());
         }
 
+        [Fact]
+        public void MultiThreadedHistogramUpdateAndSnapShotTest()
+        {
+            var boundaries = Array.Empty<double>();
+            var histogramPoint = new MetricPoint(this.aggregatorStore, AggregationType.Histogram, null, boundaries, Metric.DefaultExponentialHistogramMaxBuckets, Metric.DefaultExponentialHistogramMaxScale);
+            var argsToThread = new ThreadArguments
+            {
+                HistogramPoint = histogramPoint,
+                MreToEnsureAllThreadsStart = new ManualResetEvent(false),
+            };
+
+            var numberOfThreads = 10;
+            var snapshotThread = new Thread(HistogramSnapshotThread);
+            Thread[] updateThreads = new Thread[numberOfThreads];
+            for (int i = 0; i < numberOfThreads; ++i)
+            {
+                updateThreads[i] = new Thread(HistogramUpdateThread);
+                updateThreads[i].Start(argsToThread);
```

Comment:
Create the threads (suspend) first, then start the threads. This will increase the chance of race conditions (since thread creation is very time consuming).

Reply:
Never mind, I saw a more complex approach later (which uses `mreToEnsureAllThreadsStart` to coordinate the start).

Reply:
I guess a more stepping back question is: how likely this test will catch a race condition if the code is indeed buggy?

Reply:
> This test is not trying to catch a race condition.

I think this test would be more valuable if it could actually catch the race condition induced errors when the code has that bug.

Reply:
> I think this test would be more valuable if it could actually catch the race condition induced errors when the code has that bug.

The same test will become flaky in 1.3.0:
Check out: https://github.com/open-telemetry/opentelemetry-dotnet/compare/main-1.3.0...Yun-Ting:yunl/1.3.0-UpdateAndSnapshotWithCoyote?expand=1

I'll collect  data like below to share: waiting for an API (SpinWait) to be supported in Coyote to get the exact numbers.
![image](https://user-images.githubusercontent.com/16979322/230665093-9e5efa02-1ae4-457a-8688-add6bd5629c5.png)

Reply:
https://github.com/microsoft/coyote/pull/469